packages/mobile/src/components/lineup-tile/LineupTileActionButtons.tsx

```diff
@@ -65,88 +48,24 @@ const createStyles = (themeColors: ThemeColors) =>
     }
   })
 
-export const TrackTileActionButtons = ({
+export const LineupTileActionButtons = ({
   disabled,
   hasReposted,
   hasSaved,
   isOwner,
   isShareHidden,
   isUnlisted,
-  trackId
+  onPressOverflow,
+  onPressRepost,
+  onPressSave,
+  onPressShare
 }: Props) => {
   const { neutralLight4 } = useThemeColors()
   const styles = useThemedStyles(createStyles)
 
-  const dispatchWeb = useDispatchWeb()
-
-  const onPressOverflow = useCallback(() => {
-    if (trackId === undefined) {
-      return
-    }
-    const overflowActions = [
-      !isOwner
-        ? hasReposted
-          ? OverflowAction.UNREPOST
-          : OverflowAction.REPOST
-        : null,
-      !isOwner
-        ? hasSaved
-          ? OverflowAction.UNFAVORITE
-          : OverflowAction.FAVORITE
-        : null,
-      OverflowAction.SHARE,
-      OverflowAction.ADD_TO_PLAYLIST,
-      OverflowAction.VIEW_TRACK_PAGE,
-      OverflowAction.VIEW_ARTIST_PAGE
-    ].filter(Boolean) as OverflowAction[]
-
-    dispatchWeb(
-      openOverflowMenu({
-        source: OverflowSource.TRACKS,
-        id: trackId,
-        overflowActions
-      })
-    )
-  }, [trackId, dispatchWeb, hasReposted, hasSaved, isOwner])
-
-  const onPressShare = useCallback(() => {
-    if (trackId === undefined) {
-      return
-    }
-    dispatchWeb(
-      requestOpenShareModal({
-        type: 'track',
-        trackId,
-        source: ShareSource.TILE
-      })
-    )
-  }, [dispatchWeb, trackId])
-
-  const onToggleSave = useCallback(() => {
```

Comment:
i think this makes sense to take out, though could we keep if we just pass down the favorite source?

Reply:
Nah because it dispatches a different action (saving collection instead of track)

Reply:
ah i see, could we do something similar with lineups passing a bunch of actions, or do we want to avoid that pattern?

Reply:
Yeah but then we have to pass multiple things down, I'd rather define the function at the parent level at that point